### PR TITLE
Add a README to the src directory

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
It explains the basic build process as well as some common targets of
`build.d`. Additionally links to the source code guide to make it more
accessible.

---

Probably missed some important topics, so suggestions are welcome.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Refer to [this file](dmd/README.md) for an overview of the actual source files.
+
+## Building the compiler
+
+The compiler can be built using the integrated build system `build.d`.
+
+
+```console
+./build.d
+```
```
Comment (WalterBright) [inline src/README.md:13]: List the requirements for this to work - what version of dmd to be installed and on the path.

File: src/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+See the output of `./build.d --help` to get a list of all supported targets
+or see the list of major targets below.
+
+### Bootstrapping
+
+You can also bootstrap using `bootstrap.sh` if you don't currently have a D
```
Comment (WalterBright) [inline src/README.md:20]: In general, do not use "you" in directions. For example,

"Use `bootstrap.sh` to bootstrap the compiler if a D compiler is not currently installed."

File: src/README.md (hunk 1 above)
Comment (MoonlightSentinel) [reply src/README.md:12]: Done

**Review by WalterBright — APPROVED** [review]

**Review by wilzbach — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**